src/java/org/apache/cassandra/db/compaction/CompactionManager.java

```diff
@@ -225,6 +228,22 @@ public static void storeControllerConfig()
         }
     }
 
+    public static void publishMetrics()
+    {
+        for (String keyspace : Schema.instance.getKeyspaces())
+        {
+            // don't publish metrics for system tables
+            if (SchemaConstants.isSystemKeyspace(keyspace))
+            {
+                continue;
+            }
+            for (ColumnFamilyStore cfs : Schema.instance.getKeyspaceInstance(keyspace).getColumnFamilyStores())
+            {
+                cfs.publishMetrics();
```

Comment:
This looks like a non-trivial amount of work just to list all tables, even if we aren't really doing anything. Could we skip the work earlier (e.g. put the whole loop or even the task scheduling under a flag or a specified period that defaults to never)?